sns_driver_test.go

```diff
@@ -0,0 +1,129 @@
+package es_test
+
+import (
+	"encoding/json"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/indebted-modules/es"
+	"github.com/stretchr/testify/suite"
+)
+
+type SNSNotifierSuite struct {
+	suite.Suite
+	snsSvc   *sns.SNS
+	sqsSvc   *sqs.SQS
+	topicArn *string
+	queueURL *string
+}
+
+func TestSNSNotifierSuite(t *testing.T) {
+	suite.Run(t, new(SNSNotifierSuite))
+}
+
+func (s *SNSNotifierSuite) SetupSuite() {
+	snsEndpoint := "http://localstack:4575"
+	sqsEndpoint := "http://localstack:4576"
+	s.Eventually(func() bool {
+		_, err1 := http.Get(snsEndpoint)
+		_, err2 := http.Get(sqsEndpoint)
+		return err1 == nil && err2 == nil
+	}, 10*time.Second, time.Second, "Localstack services are not ready or running")
+
+	sess := session.Must(session.NewSession())
+	cred := credentials.NewStaticCredentials("id", "secret", "token")
+	s.snsSvc = sns.New(sess, aws.NewConfig().WithRegion("ap-southeast-2").WithCredentials(cred).WithEndpoint(snsEndpoint))
+	topicResponse, err := s.snsSvc.CreateTopic(&sns.CreateTopicInput{Name: aws.String("test-topic")})
+	s.NoError(err)
+	s.topicArn = topicResponse.TopicArn
+
+	s.sqsSvc = sqs.New(sess, aws.NewConfig().WithRegion("ap-southeast-2").WithCredentials(cred).WithEndpoint(sqsEndpoint))
+	queueResponse, err := s.sqsSvc.CreateQueue(&sqs.CreateQueueInput{QueueName: aws.String("test-queue")})
+	s.NoError(err)
+	s.queueURL = queueResponse.QueueUrl
+
+	_, err = s.snsSvc.Subscribe(&sns.SubscribeInput{
+		Protocol: aws.String("sqs"),
+		Endpoint: queueResponse.QueueUrl,
+		TopicArn: topicResponse.TopicArn,
+	})
```

Comment:
Yep we can subscribe a queue to the topic for testing, but also directly via SDK to avoid more dependencies: https://docs.aws.amazon.com/sdk-for-go/api/service/sns/#SNS.Subscribe

Reply:
This is the SDK API I used. Unless I'm missing something, the `SubscriptionInput` requires an `Endopoint` (https://docs.aws.amazon.com/sdk-for-go/api/service/sns/#SubscribeInput), and among the available options, SQS seemed the most straight forward to support testing…